src/nvim/option.c

```diff
@@ -301,6 +301,15 @@ static char *(p_cot_values[]) = {"menu", "menuone", "longest", "preview",
 # include "option.c.generated.h"
 #endif
 
+static void set_runtimepath_default(void)
+{
+  garray_T rtp_ga;
+  ga_init(&rtp_ga, (int)sizeof(const char *), 1);
+  GA_APPEND(const char *, &rtp_ga, get_user_conf_dir());
+  GA_APPEND(const char *, &rtp_ga, concat_fnames(get_user_conf_dir(), "after", true));
+  set_string_default("runtimepath", ga_concat_strings(&rtp_ga));
```

Comment:
Unfortunately (I guess), we have a lot of code that check paths are less than the maximum, so this convention wouldn't be easy to remove from the codebase, even if it's wrong. For the moment, I think it would be correct to just say "we don't support unreasonably large paths," and if supporting them means unleashing such inefficient code such as in the linked article, I don't see any benefit.

Also, I double checked: I should have said `MAXPATHL`, an os-independent variable that we define.